sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/models/ThreadCreationDateRetentionPolicy.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.communication.chat.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.json.JsonReader;
+import com.azure.json.JsonToken;
+import com.azure.json.JsonWriter;
+
+import java.io.IOException;
+
+/**
+ * Thread retention policy based on thread creation date.
+ */
+@Fluent
+public final class ThreadCreationDateRetentionPolicy extends ChatRetentionPolicy {
+    /*
+     * Retention Policy Type
+     */
+    private final com.azure.communication.chat.models.RetentionPolicyKind kind
+        = com.azure.communication.chat.models.RetentionPolicyKind.THREAD_CREATION_DATE;
+
+    /*
+     * Indicates how many days after the thread creation the thread will be deleted.
+     */
+    private int deleteThreadAfterDays;
+
+    /**
+     * Creates an instance of ThreadCreationDateRetentionPolicy class.
+     */
+    public ThreadCreationDateRetentionPolicy() {
+    }
+
+    /**
+     * Get the kind property: Retention Policy Type.
+     *
+     * @return the kind value.
+     */
+    @Override
+    public com.azure.communication.chat.models.RetentionPolicyKind getKind() {
+        return this.kind;
+    }
+
+    /**
+     * Get the deleteThreadAfterDays property: Indicates how many days after the thread creation the thread will be
+     * deleted.
+     *
+     * @return the deleteThreadAfterDays value.
+     */
+    public int getDeleteThreadAfterDays() {
+        return this.deleteThreadAfterDays;
+    }
+
+    /**
+     * Set the deleteThreadAfterDays property: Indicates how many days after the thread creation the thread will be
+     * deleted.
+     *
+     * @param deleteThreadAfterDays the deleteThreadAfterDays value to set.
+     * @return the ThreadCreationDateRetentionPolicy object itself.
+     */
+    public com.azure.communication.chat.models.ThreadCreationDateRetentionPolicy
+        setDeleteThreadAfterDays(int deleteThreadAfterDays) {
```

Comment:
Why are we seeing so many qualified names in these public models? This one is from this same class 😮